### PR TITLE
Obsolete ADLER-32 but don't forbid it. It's now NOT RECOMMENDED #828 

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

Obsolete ADLER-32 but don't forbid it. #828

### CODE REVIEW

**Comment by ioggstream** [comment]
@manger will this work for you?

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -760,6 +768,20 @@ registry:
...
+Algorithm
+Values](https://www.iana.org/assignments/http-dig-alg/http-dig-alg.xhtml)
+registry:
+
+* Digest Algorithm: ADLER32
+* Description: The ADLER32 algorithm is a checksum specified in [RFC1950] "ZLIB Compressed Data Format".
+  The 32-bit output is encoded in hexadecimal (using between 1 and 8 ASCII characters from 0-9, A-F, and a-f; leading 0's are allowed).
+  For example, ADLER32=03da0195 and ADLER32=3DA0195 are both valid checksums for the 4-byte message "Wiki".
+  This algorithm is obsoleted and SHOULD NOT be used.
+* Status: obsoleted
```
Comment (ioggstream) [inline draft-ietf-httpbis-digest-headers.md]: Add `Reference` to this document.

**Comment by paulmillar** [comment]
I think the pull request is OK as-is, but might be improved slightly.

In particularly, I'm wondering if SHOULD NOT is too strong for the server.  A server could support ADLER32 without that support impacting on cryptographic data integrity provided that server also supports a cryptographically secure hash; e.g., SHA-512.  Clients that wish to protect against malicious actor changes would use the SHA-512 hash and not request (or ignore) the ADLER32 value.  Therefore, I would suggest that, for the server, the document says a server MAY implement ADLER32.

I think it is perfectly reasonable for the document to say the client SHOULD NOT use ADLER32, for the reasons stated.

That said, I can't decide if this distinction is really helpful, so I defer to your judgement, here.

**Comment by LPardue** [comment]
I'm not so sure we want to make a distinction between client and server as you write it. Both client and server can attach Digest header to request or response. It is the receiver of Digest that is exposed to the risk of using a weak digest.

The current language avoids mentioning either endpoint so we leave it to implementers. I'm not opposed to spelling things out more clearly but we'd really need to nail down what we mean by "using Digest  algorithms" because that relates to requesting them, generating them, regenerating them (proxy), validating them (endpoint or proxy) etc.

**Comment by paulmillar** [comment]
That's a good point about client/server.  Sorry, I had that wrong.  As you say, it's the recipient of a Digest that must decide whether the algorithm is sufficient.  This recipient may be either the client or the server.

I think the document is right to point out the dangers of accepting a weak digest.

Perhaps there is a useful distinction between generating a digest and accepting a digest; e.g., an agent MAY support generating a weak digest but SHOULD NOT accept a weak digest.

**Comment by LPardue** [comment]
> That's a good point about client/server. Sorry, I had that wrong.

Not wrong, just partially true. 

> Perhaps there is a useful distinction between generating a digest and accepting a digest; e.g., an agent MAY support generating a weak digest but SHOULD NOT accept a weak digest.

That could work for me. I'll leave it to @ioggstream to decide what he things makes sense to do.

**Comment by ioggstream** [comment]
@paulmillar @LPardue as this issue is for ADLER, I moved the `weak digest` discussion  to #894

**Comment by ioggstream** [comment]
Merging as for the positive feedback from @phluid61 and @LPardue comment https://github.com/httpwg/http-extensions/pull/877#issuecomment-520372044

**Pull request merged**